active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,660 @@
+
+### Preferred placement
+
+Teaching tip replicates all of Flyout's [FlyoutPlacementMode](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutPlacementMode) placement behavior with the TeachingTipPlacementMode property. The default placement mode will try to place a pointing teaching tip above its target and a non-pointing teaching tip in the bottom right corner of the UWP window. As with Flyout, if the preferred placement mode would not leave room for the teaching tip to show, another placement mode will be automatically chosen. 
+
+A pointing teaching tip with its PreferredPlacement set to "BottomEdgeAlignedLeft" will appear below its target and will have its left edge aligned to the left edge of the target.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" Click="OnFirstSaveButtonClick">
+    <Button.Resources>
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Target="{x:Bind SaveButton}"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            PreferredPlacement="BottomEdgeAlignedLeft">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    AutoSaveTip.IsOpen=true;
+}
+```
+
+![A sample app with a teaching tip pointing upward at a button that reads "Save". The left edge of the teaching tip and the "Save" button are aligned. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipPointingPreferredPlacementSample.jpg)
+
+
+A non-pointing teaching tip with its PreferredPlacement set to "BottomEdgeAlignedLeft" will appear in the bottom left corner of the UWP window.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" Click="OnFirstSaveButtonClick">
+
+<muxc:TeachingTip x:Name="AutoSaveTip"
+    Title="Saving automatically"
+    Subtitle="We save your changes as you go - so you never have to."
+    PreferredPlacement="BottomEdgeAlignedLeft">
+</muxc:TeachingTip>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    AutoSaveTip.IsOpen=true;
+}
+```
+
+![A sample app with a teaching tip in the bottom left corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipNonPointingPreferredPlacementSample.jpg)
+
+### Escaping UWP window bounds
+
+On Windows version 19H1 and above, a teaching tip can escape UWP window bounds by setting the ShouldConstrainToRootBounds property. On earlier versions of Windows, this property is ignored and the teaching tip always stays within the UWP window bounds.
```

Comment:
>On earlier versions of Windows, this property is ignored and the teaching tip always stays within the UWP window bounds. [](start = 132, length = 120)

On earlier versions of windows this property doesn't exist. We should probably have the sample be conditional markup like contract9present:ShouldConstrainToRootBounds="true".

Reply:
@jevansaks, I checked with @StephenLPeters and this property is being surfaced by teaching tip, so I believe it is always present even if the underlying popup doesn't have the property.

Reply:
More specifically (from @StephenLPeters):

ShouldConstrainToRootBounds is a property on TeachingTip.  It will be present for every TeachingTip that can exist. There happens to be (by design) a property on Popup called ShouldConstrainToRootBounds, this proprety may or may not exist in the context of the TeachingTip in question.  If this popup property is not present, for whatever reason, the property on TeachingTip won't do anything.  If the proprety on popup is present, then setting this proprety on teaching tip will result in the value being set on the TeachingTip's internal popup.  In both cases the property on TeachingTip is present.